src/checkers/inference/solver/backend/z3smt/Z3SmtFormatTranslator.java

```diff
@@ -0,0 +1,90 @@
+package checkers.inference.solver.backend.z3smt;
+
+import checkers.inference.model.ArithmeticVariableSlot;
+import checkers.inference.model.CombVariableSlot;
+import checkers.inference.model.ConstantSlot;
+import checkers.inference.model.ExistentialVariableSlot;
+import checkers.inference.model.LubVariableSlot;
+import checkers.inference.model.RefinementVariableSlot;
+import checkers.inference.model.Slot;
+import checkers.inference.model.SourceVariableSlot;
+import checkers.inference.solver.backend.AbstractFormatTranslator;
+import checkers.inference.solver.backend.z3smt.encoder.Z3SmtSoftConstraintEncoder;
+import checkers.inference.solver.frontend.Lattice;
+import com.microsoft.z3.BoolExpr;
+import com.microsoft.z3.Context;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.AnnotationMirror;
+
+// AbstractFormatTranslator<SlotEncodingT, ConstraintEncodingT, SlotSolutionT>
+public abstract class Z3SmtFormatTranslator<SlotEncodingT, SlotSolutionT>
+        extends AbstractFormatTranslator<SlotEncodingT, BoolExpr, SlotSolutionT> {
+
+    protected Context ctx;
+
+    /** Cache of all serialized slots, keyed on slot ID. */
+    protected final Map<Integer, SlotEncodingT> serializedSlots;
```

Comment:
I can't find any usage of this field in the PR.

Reply:
This field is used in the subtype of `Z3SmtFormatTranslator` in value inference.